### PR TITLE
rt: 5.0.3 -> 5.0.5

### PR DESCRIPTION
**Pull request by dasJ**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by stigtsp — COMMENTED** [review]
nixpkgs-review gives me this:

```
[..]
       error: attribute 'getOut' missing

       at /home/sgo/.cache/nixpkgs-review/pr-262283-1/nixpkgs/pkgs/top-level/perl-packages.nix:10753:26:

        10752|     HOME = "/build";
        10753|     FONTCONFIG_PATH = "${lib.getOut pkgs.fontconfig}/etc/fonts";
             |                          ^
        10754|

```

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -10732,7 +10732,29 @@ with self; {
...
+      hash = "sha256-h8hcbt/86k+W5rSAD2+VEq6rGeuNOzSDAachMxvLhYA=";
+    };
+
+    # XXX: It'd be nicer if `GraphViz.pm' could record the path to graphviz.
+    buildInputs = [ pkgs.graphviz TestPod Moo IPCRun3 TypeTiny TestSnapshot Graph ];
+    propagatedBuildInputs = [ FileWhich IPCRun ParseRecDescent XMLTwig XMLXPath DataSectionSimple ];
+
+    # needed for fontconfig tests
+    HOME = "/build";
+    FONTCONFIG_PATH = "${lib.getOut pkgs.fontconfig}/etc/fonts";
```
Comment (zakame) [inline pkgs/top-level/perl-packages.nix:10753]: Maybe this should be 

```suggestion
    FONTCONFIG_PATH = "${pkgs.fontconfig.out}/etc/fonts";
```

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -6257,12 +6257,12 @@ with self; {
...
+    version = "0.07";
     src = fetchurl {
-      url = "mirror://cpan/authors/id/A/AL/ALEXMV/Date-Extract-0.06.tar.gz";
-      hash = "sha256-vHZY1cUMNSXsDvy1Ujal3i1dT8BvwUf6OSnI8JU82is=";
+      url = "mirror://cpan/authors/id/E/ET/ETHER/Date-Extract-0.07.tar.gz";
+      hash = "sha256-+geIBK3k7uwd4UcuDguwR65i5MjU1QIHAbnlBXfFuPQ=";
     };
-    buildInputs = [ TestMockTime ];
+    buildInputs = [ TestMockTime TestMockTimeHiRes ];
     propagatedBuildInputs = [ DateTimeFormatNatural ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:6266]:
```suggestion
    buildInputs = [ TestMockTimeHiRes ];
    propagatedBuildInputs = [ ClassDataInheritable DateTimeFormatNatural ];
```

.. so we don't get a conflict with d6efde33dc80023d8976c2a4cdeac1440554d141

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -98,7 +98,7 @@ with self; {
       hash = "sha256-6nyqFPdX3ggzEO0suimGYd3Mpd7gbsjxgEPqYlp53yA=";
     };
 
-    outputs = ["out" "man"];
+    outputs = [ "out" "man" ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:101]: Unrelated change?

File: pkgs/top-level/perl-packages.nix [hunk 4]
```diff
@@ -9222,7 +9222,7 @@ with self; {
     propagatedBuildInputs = [ ModulePluggable Moo TypeTiny namespaceautoclean ];
     meta = {
       description = "Watch for changes to files, cross-platform style";
-      license = with lib.licenses; [artistic2 ];
+      license = with lib.licenses; [ artistic2 ];
```
[the remark this answers is not in the document]
Comment (dasJ) [reply pkgs/top-level/perl-packages.nix:9225]: These were actually intentional. Should I push the single commit (first one) into a separate PR instead?

[the remark this answers is not in the document]
Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix:9225]: Nah, didn't notice the commit :)

File: pkgs/top-level/perl-packages.nix (hunk 2 above)
Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix]: Sorry, this also depends on 30975d969998a6ec618a4d15768a2ac3412592fe which adds `propagatedBuildInputs = [ TestMockTime ];` to `TestMockTimeHiRes`.

You can add back `TestMockTime` as a buildInput here and I'll sort it out when it reaches staging-next.

Comment (stigtsp) [reply pkgs/top-level/perl-packages.nix]: @dasJ Added a PR to this PR here: https://github.com/helsinki-systems/nixpkgs/pull/3

**Comment by risicle** [comment]
Assume the security issues are CVE-2023-41259 & CVE-2023-41260?

This package is missing a _lot_ of `meta` information too.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64.

**Review by stigtsp — APPROVED** [review]
LGTM

**Pull request merged**